aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -946,14 +1008,33 @@ Tensor std(const Tensor& self, bool unbiased) {
               "std only supports CPU AND CUDA device type, got: ", self.device().type());
   TORCH_CHECK(self.layout() == Layout::Strided,
               "std only supports strided layout, got: ", self.layout());
-  TORCH_CHECK(at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type()),
-              "std only supports floating-point dtypes");
   auto trivial_return = _allreduce_return_trivial(self, std::numeric_limits<double>::quiet_NaN());
+
+  // Pre-casts integral inputs to the default scalar type
+  // Note: _th_std (_std) does not support integral types
+  auto maybe_dtype = get_floating_promotion_dtype(self, c10::nullopt);
+  if (maybe_dtype.has_value()) {
+    return trivial_return.has_value() ?
+      trivial_return.value() :
+      at::_std(self.to(*maybe_dtype), unbiased);
+  }
+
+  // Diverts complex types to tensor iterator's impl
```

Comment:
how did this work before then?  This function wasn't calling std_var_out?

Reply:
It did not work before. It would throw a RunTime error because it did not call std_var_out in this case.